sdk/core/azure-core/test/ut/transport_adapter.cpp

```diff
@@ -439,4 +439,59 @@ namespace Azure { namespace Core { namespace Test {
     CheckBodyFromBuffer(*response, expectedResponseBodySize);
   }
 
+  TEST_F(TransportAdapter, cancelTransferUpload)
+  {
+    Azure::Core::Http::Url host("http://httpbin.org/put");
+    Azure::Core::Context cancelThis;
+
+    auto threadRoutine = [host, cancelThis]() {
+      // Start a big upload and expect it to throw cancelation
+      std::vector<uint8_t> bigBuffer(1024 * 1024 * 200, 'x'); // upload 200 Mb
+      auto stream = Azure::Core::Http::MemoryBodyStream(bigBuffer);
+      auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Put, host, &stream);
+
+      // Request will be canceled from main thread throwing the excpetion
+      EXPECT_THROW(pipeline.Send(cancelThis, request), Azure::Core::RequestCanceledException);
+    };
+
+    // Start request
+    std::thread t1(threadRoutine);
+
+    // Wait 2 sec so we know upload has started
+    std::this_thread::sleep_for(std::chrono::milliseconds(2000));
+
+    // Cancel
+    cancelThis.Cancel();
+
+    // join job before return
+    t1.join();
+  }
+
+  TEST_F(TransportAdapter, cancelTransferDownload)
+  {
+    // 50Mb text
+    Azure::Core::Http::Url host("https://raw.githubusercontent.com/vhvb1989/"
+                                "azure-sdk-for-c-samples/master/filesRepository/fortyMb");
```

Comment:
Is it 50 mb or 40?

Reply:
Also, is there a way for us to check this in (or generate it programmatically), so that tests don't rely on your personal, user-specific file link?

Reply:
@weshaggard - we want to download some big payload from a url/server to test request cancellation. Do you know of a good way to do this as part of basic unit tests (without needed to setup a storage client/upload or the live testing approach)?

Can we upload this file to an azure sdk bot GitHub account?

Reply:
move to https://github.com/Azure/azure-sdk-for-cpp/pull/782